server/src/test/java/org/elasticsearch/snapshots/SnapshotsInProgressSerializationTests.java

```diff
@@ -187,44 +187,8 @@ protected Custom mutateInstance(Custom instance) {
     }
 
     private Entry mutateEntry(Entry entry) {
-        switch (randomInt(8)) {
```

Comment:
These examples were not legal state transitions for a snapshot entry so I didn't implement diffing for them and remove the test cases.

Reply:
Hmm but we also use `mutateEntry` to test the non-diff serialization, and equals/hashcode, and in that case we do want to verify these mutations.

Reply:
Ah right it's also one of those tests. I pushed https://github.com/elastic/elasticsearch/pull/89619/commits/852cf9d818db9be81c7e13fe01095f8bd4973371to split out the "legal" mutations so I can just use them for the diffing and brought the other ones back.